### PR TITLE
Final step to remove IPD (remove _ccpp labels from regression tests), RAS updates and regression tests

### LINKED ISSUE #214: Remove IPD
## Description

Now that we have a GFS v16 reference tag for the ufs-weather-model develop branch and its submodules, it is time to remove IPD.

## Solution

This should be done in several steps.

1. Remove all source files that are required for IPD only (in fv3atm) and make `CCPP=Y` the default in the ufs-weather-model compile/build scripts. This should not change the regression test results. However, since `CCPP=Y` is no longer in the MAKEOPT string, the suffices `_prod` and `_ccpp` for the regression test baseline/run directories will disappear (hence the requirement to create new baselines).

2. Move the computation of snow cover and other variables from `io/FV3GFS_io.F90` into CCPP (probably  the `time_vary` group), and remove the `soilveg` initialization in `GFS_driver.F90`. This should not change the regression test results when the fv3atm and the dycore are compiled in double precision (64bit), but it will change the results when fv3atm and the dycore are compiled in single precision (32bit). This is because the computation fo snow cover and other variables in `FV3GFS_io.F90` uses the same precision as fv3atm and the dycore, whereas it is always double precision in the CCPP physics.

3. Remove all `#ifndef CCPP` and related preprocessor directives from fv3atm and the ufs-weather-model, but not GFDL_atmos_cubed_sphere (These will still be needed for compatibility with how GFDL is running physics? To be confirmed with @bensonr). Remove definition of IPD data types in fv3atm and replace use of IPD data types with GFS data types. This should not change the regression test results. Update: the preprocessor directives can also be removed from the dycore.

4. Rename all regression tests (drop `_ccpp` from `fv3_ccpp_xyz`). All non-CCPP `*_run.IN` templates in `tests/fv3_conf` should be removed, and all CCPP templates renamed (drop `ccpp_` from `ccpp_xyz_run.IN`). Drop suffices `_ccpp`/`_prod` from regression test baseline/run directories (but need to keep repro? add `_debug` in the same way? or manually change the regression test control directories for `REPRO=Y` to `_repro`?)

5. Clean up all `CMakeLists.txt` files that contain legacy code to support building IPD and CCPP, using a top-level cmake or gnumake build. Now we only need to support CCPP with a cmake build.

6. Anything that is missing in the previous bullets.

## Alternatives

No alternative.

## Testing:

Testing will be done on all tier-1,2,3 platforms (tier-1 pre-commit, tier-2,3 as time permits).

## Dependent PRs:

Step 1 completed 12/29/2020 without removing the suffices `_prod` and `_ccpp`, see https://github.com/ufs-community/ufs-weather-model/pull/331.

Step 3 completed 01/14/2021, see https://github.com/ufs-community/ufs-weather-model/pull/357 and associated PRs.

Step 5 completed 01/14/2021, see https://github.com/ufs-community/ufs-weather-model/pull/357 and associated PRs.

Parts of step 4 are addressed in https://github.com/ufs-community/ufs-weather-model/pull/378.

Parts of step 2 are addressed in https://github.com/ufs-community/ufs-weather-model/pull/407.

### PR DESCRIPTION
**Pull request by climbfuji**
## Description

This PR contains the following changes:
- add new regression test for RAS
- remove `_ccpp` label from all regression test names, regression test setup scripts, namelist templates and the `rt.sh` run directories
- update submodule pointer for RAS changes and other changes described in https://github.com/NOAA-EMC/fv3atm/pull/252 and https://github.com/NCAR/ccpp-physics/pull/585

This PR completes the removal of IPD.

### Issue(s) addressed

Fixes https://github.com/NOAA-EMC/fv3atm/issues/214

## Testing

Regression testing will be done on all tier-1 platforms.

New baselines required because of the name changes of the directories. New results for RAS tests.

Regression testing completed on:
- cheyenne.gnu
- cheyenne.intel
- gaea.intel
- hera.gnu
- hera.intel
- jet.intel

Note:
- wcoss is under maintenance for the rest of this week
- orion is extremely busy, essentially unusable; baseline creation is still ongoing

## Dependencies

https://github.com/NCAR/ccpp-physics/pull/585
https://github.com/NOAA-EMC/fv3atm/pull/252
https://github.com/ufs-community/ufs-weather-model/pull/448

### CODE REVIEW

File: tests/parm/rasmgshoc.nml.IN [hunk 1]
```diff
@@ -0,0 +1,355 @@
+&amip_interp_nml
+       interp_oi_sst = .true.
+       use_ncep_sst = .true.
+       use_ncep_ice = .false.
+       no_anom_sst = .false.
+       data_set = 'reynolds_oi',
+       date_out_of_range = 'climo',
+/
```
Comment (DusanJovic-NOAA) [inline tests/parm/rasmgshoc.nml.IN]: This namelist `amip_interp_nml` is not used in ufs-weather-model (fv3atm). At some point (maybe even in this PR) we should remove it from all input namelist (templates). Few other namelists like `coupler_nml` and `interpolator_nml`
 should also be removed.

**Review by SMoorthi-emc — APPROVED** [review]
We may have to make some changes when Model*ntchm > 0 and RAS is used as the default fscav is changed to 0.6 (RAS assumes fscav that enters RAS has values for all tracers, including tke). Also, scavenging in RAS itself may have to be updated at that point.

**Comment by climbfuji** [comment]
This PR is ready for final review, I kicked off the CI tests. Orion is still busy creating baselines, after more than 24h.

GitHub does a terrible job with displaying the diffs in this commit, even though I used `git mv` it thinks that I deleted the CCPP regression test file (run script, namelist template) and manually updated the non-CCPP version (if present).

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Comment by junwang-noaa** [comment]
CI test failed, it will be fixed in next PR. 
Orion is busy, and wcoss is not available, we still need to make up the baselines when these machines are available.

**Comment by climbfuji** [comment]
Baseline creation and verification of results finished post-commit on orion. Attaching log here.

[RegressionTests_orion.intel.log](https://github.com/ufs-community/ufs-weather-model/files/6173002/RegressionTests_orion.intel.log)

**Pull request merged**